### PR TITLE
Feature/yaml error handling

### PR DESCRIPTION
**Pull request by live1206**
Pass along Yaml syntax error for TOC loading and add e2e test for it. Throw exception for Config loading.

### CODE REVIEW

File: src/docfx/build/toc/parser/TableOfContentsParser.cs [hunk 1]
```diff
@@ -72,11 +66,11 @@ private static List<TableOfContentsInputItem> LoadTocModel(string content, strin
                 {
                     throw new NotSupportedException($"{filePath} is not a valid TOC file, detail: {ex.Message}.", ex);
```
Comment (OsmondJiang) [inline src/docfx/build/toc/parser/TableOfContentsParser.cs]: >throw new NotSupportedException($"{filePath} is not a valid TOC file, detail: {ex.Message}.", ex); [](start = 20, length = 98)

also move this to error?

File: src/docfx/config/Config.cs [hunk 2]
```diff
@@ -150,7 +151,12 @@ private static Config LoadCore(string configPath, CommandLineOptions options = n
         private static JObject LoadOriginalConfigObject(string configPath, List<string> parents, bool expand)
         {
             // TODO: support URL
-            var (_, config) = YamlUtility.Deserialize<JObject>(File.ReadAllText(configPath));
+            var (errors, config) = YamlUtility.Deserialize<JObject>(File.ReadAllText(configPath));
+            if (errors.Any())
```
Comment (OsmondJiang) [inline src/docfx/config/Config.cs]: >errors.Any() [](start = 16, length = 12)

will it be null?

File: docs/specs/toc.yml [hunk 3]
```diff
@@ -792,4 +792,22 @@ outputs:
   build.log: |
     ["error","invalid-toc-href","The toc href 'f1/a.md' can only reference to a local TOC file, folder or absolute path","docs/toc.yml"]
     ["error","invalid-toc-href","The toc href 'f1/a.md' can only reference to a local TOC file, folder or absolute path","docs/toc.yml"]
-
+---
+# Toc syntax error with duplicated keys
+inputs:
+  docfx.yml:
+  docs/toc.yml: |
+    - name: topic href has highest priority
```
Comment (OsmondJiang) [inline docs/specs/toc.yml]: >topic href has highest priority [](start = 12, length = 31)

change the example name

File: docs/specs/toc.yml [hunk 4]
```diff
@@ -792,4 +792,22 @@ outputs:
...
+      topicHref: f1/a.md
+      topicHref: f1/a.md
+      tocHref: f1/
+      href: f1/b.md
+  docs/f1/a.md:
+  docs/f1/b.md:
+outputs:
+  docs/toc.json:
```
Comment (OsmondJiang) [inline docs/specs/toc.yml]: >json [](start = 11, length = 4)

what's the output toc looks like?

File: src/docfx/config/Config.cs (hunk 2 above)
Comment (live1206) [reply src/docfx/config/Config.cs]: nope, will  return an empty list if no error

File: docs/specs/toc.yml (hunk 4 above)
Comment (live1206) [reply docs/specs/toc.yml]: the actual output toc is an empty list: 
{"items":[]}

File: src/docfx/build/toc/parser/TableOfContentsParser.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/build/toc/parser/TableOfContentsParser.cs]: ?

---
In reply to: [198069924](https://github.com/dotnet/docfx/pull/2943#discussion_r198069924) [](ancestors = 198069924)

**Pull request merged**